release-content/0.14/release-notes/13417_implement_the_full_set_of_sort_methods_on_QueryIter.md

```diff
@@ -1 +1,114 @@
-TODO
+Currently, query iteration order is not guaranteed. If we wish to work with our query items in a certain order, we need sorting.
+We might want sort our queries to f.e. display a list of things, or fold over our query with some order-dependent math operation.
+In 0.13 a sort could look like this:
+
+```rust
+#[derive(Component, Copy, Clone, Deref)]
+pub struct Attack(pub usize)
+
+fn handle_enemies(enemies: Query<(&Health, &Attack, &Defense)>) {
+    let mut enemies: Vec<_> = enemies.iter().collect();
+    enemies.sort_by_key(|(_, atk, ..)| *atk)
+    for enemy in enemies {
+        work_with(enemy)
+    }
+}
+```
+
+This can get especially unwieldy and repetitive when sorting within multiple systems.
+Even if we always want the same sort, different [`Query`] types make it unreasonably difficult to abstract away as a user!
+To solve this, we implemented new sort methods on the [`QueryIter`] type, turning the example into:
+
+```rust
+// To be used as a sort key, `Attack` now implements Ord.
+#[derive(Component, Copy, Clone, Deref, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Attack(pub usize)
+
+fn handle_enemies(enemies: Query<(&Health, &Attack, &Defense)>) {
+    for enemy in enemies.iter().sort::<&Attack>() {
```

Comment:
Does this also allocate under the hood? If so, we should call it out.

Reply:
A call-out has been added, (without mentioning the Vec type)